### PR TITLE
Cleanup after compat test

### PR DESCRIPTION
**Pull request by erikbosch**
Previously out files were not removed. Also some minor refactoring

### CODE REVIEW

**Comment by erikbosch** [comment]
@nwesem feel free to review

Fixes #353

**Review by nwesem — COMMENTED** [review]
we haven't been doing this but could be type hinted like this

```
...
def change_test_dir(request: pytest.FixtureRequest, monkeypatch: pytest.MonkeyPatch):
...
```

File: tests/backward_compatibility/test_compatibility.py [hunk 1]
```diff
@@ -43,7 +50,6 @@ def test_compatibility(tag, change_test_dir):
     Test that we still can analyze wanted versions without error
     """
 
-    os.system("rm -rf vehicle_signal_specification")
     os.system("git clone --depth 1 --branch " + tag +
```
Comment (nwesem) [inline tests/backward_compatibility/test_compatibility.py:53]: you could move this to the fixture above the yield so it is only cloned once in the beginning, no?

Comment (nwesem) [reply tests/backward_compatibility/test_compatibility.py:53]: but I just realized that that would mean you would have to change the scope of the `delete_files` fixture function to `"module"`..

Comment (nwesem) [reply tests/backward_compatibility/test_compatibility.py:53]: or would it be `"package"`, I'm currently not sure https://docs.pytest.org/en/6.2.x/fixture.html

Comment (nwesem) [reply tests/backward_compatibility/test_compatibility.py:53]: and obviously switch branch depending on tag, but please ignore if this creates any overhead

Comment (erikbosch) [reply tests/backward_compatibility/test_compatibility.py:53]: Would indeed be doable. Lets see if I get time. As we only have one test case here the advantage might be limited, but we could possibly use it as playground and later reuse it for other files

Comment (erikbosch) [reply tests/backward_compatibility/test_compatibility.py:53]: I did some tests, the `monkeypatch.chdir` command has function scope, but even if refactoring like below it is faster to do 3 shallow clones for a single branch than doing a full clone. So it does not seem to be worth the effort right now

```
@pytest.fixture(scope="module", autouse=True)
def initiate_test_env(request):
    os.chdir(request.fspath.dirname)
    os.system("git clone https://github.com/COVESA/vehicle_signal_specification")
    yield None
    os.system("rm -rf out.json out.txt vehicle_signal_specification")
    os.chdir(request.config.invocation_params.dir)
```

**Pull request merged**